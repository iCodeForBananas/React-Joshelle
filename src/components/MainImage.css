.MainImage-rainbow {
  background-color: #302244;
  border: 50px solid transparent;
  border-image: linear-gradient(to bottom right, #b827fc 0%, #2c90fc 25%, #b8fd33 50%, #fec837 75%, #fd1892 100%);
  border-image-slice: 1;
  position: absolute;
  top: 50%;
  left: 50%;
  width: 500px;
  margin-top: -150px;
  margin-left: 200px;
  z-index: 1;
}

.MainImage-image {
  max-width: 100%;
  max-height: 100%;
}

.MainImage-spin {
  animation: MainImage-spin infinite 0.421s ease;
}

@keyframes MainImage-spin {
  50% {
    transform: scale(1.3, 1.3) rotate(-5deg);
  }
  100% {
    transform: scale(1, 1) rotate(5deg);
  }
}
