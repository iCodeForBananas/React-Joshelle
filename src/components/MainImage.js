// @flow
import React from "react";
import "./MainImage.css";

const MainImage = function() {
  return (
    <div className="MainImage-rainbow MainImage-spin">
      <img src="/josh.gif" className="MainImage-image" alt="" />
    </div>
  );
};

export default MainImage;
