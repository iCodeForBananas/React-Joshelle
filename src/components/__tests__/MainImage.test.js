import React from "react";
import MainImage from "../MainImage";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<MainImage />).toJSON();
  expect(tree).toMatchSnapshot();
});
