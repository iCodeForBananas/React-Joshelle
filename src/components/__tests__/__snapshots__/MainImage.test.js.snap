// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MainImage-rainbow MainImage-spin"
>
  <img
    alt=""
    className="MainImage-image"
    src="/josh.gif"
  />
</div>
`;
